prov/sm2/src/sm2_ep.c

```diff
@@ -289,12 +289,52 @@ static ssize_t sm2_do_inject(struct sm2_ep *ep, struct sm2_region *peer_smr,
 
 	sm2_generic_format(xfer_entry, ep->gid, op, tag, data, op_flags,
 			   context);
-	sm2_format_inject(xfer_entry, mr, iov, iov_count, peer_smr);
+	sm2_format_inject(xfer_entry, mr, iov, iov_count);
 
 	sm2_fifo_write(ep, peer_gid, xfer_entry);
 	return FI_SUCCESS;
 }
 
+static void sm2_format_cma(struct sm2_xfer_entry *xfer_entry,
+			   const struct iovec *iov, size_t count)
+{
+	struct sm2_cma_data *cma_data = &xfer_entry->cma_data;
+
+	/* All CMA messages must be delivery complete to stop completion from
+	 * being written instantly */
+	xfer_entry->op_flags |= FI_DELIVERY_COMPLETE;
```

Comment:
By `|=` delivery completion flag, it seems to be possible to have multiple, conflicting completion semantic flags set.  

Should we consider clearing the other completion flags?

Reply:
I don't think it matters if you give a higher completion status... i.e.. the user asked for transmit complete, and we return delivery complete.